modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java

```diff
@@ -303,7 +312,7 @@ private ClientRequestFuture send(ClientOperation op, Consumer<PayloadOutputChann
     private <T> T receive(ClientRequestFuture pendingReq, Function<PayloadInputChannel, T> payloadReader)
         throws ClientException {
         try {
-            byte[] payload = pendingReq.get();
+            byte[] payload = timeout > 0 ? pendingReq.get(timeout) : pendingReq.get();
```

Comment:
Should we do something similar in `receiveAsync`?

Reply:
`receiveAsync` calls `get()` on the future only after `pendingReq` completion.
Also, `CompletableFuture` is returned to the user, so the user can use `get()` method with a timeout to gain the same result.

Reply:
As a user, I would expect sync and async operations to behave the same when a timeout is set without extra manual steps.

Anyway, let's keep it this way for now since a refactoring in coming in [IGNITE-13496](https://issues.apache.org/jira/browse/IGNITE-13496).